dhis2/dhis2-core
# fix: update dashboard with not accessible item objects

https://dhis2.atlassian.net/browse/DHIS2-6556
Original PR https://github.com/dhis2/dhis2-core/pull/12276

### Issue
- If user update a Dashboard without METADATA_READ access to DashboardItem's objects, then those objects will be disassociate from corresponding DashboardItem. 

### Fix
- Add a new `handleDashboard` method to `PreheatService` in order to preheat all DashboardItem's objects without sharing check.
- This will only applies to version 2.39 and below. For 2.40 we will need to add validation to block user from updating the Dashboard in this use case. User then will have to be granted METADATA_READ access to all DashboardItem's objects in order to update the Dashboard. This can be achieved by cascade sharing function.

### Test
1. Login with SuperUser.
2. Create a new UserA, user should not have `ALL` authority, but can read/update Dashboard.
3. Create a new dashboard `DashboardA`, add one Visualization to it.
4. In Visualization app, update sharing settings of Visualization in step 2 to make user UserA doesn't have METADATA_READ access to it.
5. Share `DashboardA` to UserA with METADATA_READ_WRITE permission.
6. Login with UserA.
7.  In Dashboard app, edit `DashboardA` , add one more Visualization to it.
8. Expect: after saving,  `DashboardA` has two DashboardItems and two Visualizations attached to it.
